wsd/COOLWSD.cpp

```diff
@@ -1004,6 +1011,277 @@ COOLWSD::~COOLWSD()
 {
 }
 
+#if !MOBILEAPP
+
+// A custom socket poll to fetch remote config every 60 seconds
+// if config changes it applies the new config using LayeredConfiguration
+class RemoteConfigPoll : public SocketPoll
+{
+public:
+    RemoteConfigPoll(LayeredConfiguration& config)
+        : SocketPoll("remoteconfig_poll")
+        , conf(config)
+    {
+        remoteServerURI = Poco::URI(conf.getString("remote_config.remote_url"));
+    };
+
+    void start()
+    {
+        if (remoteServerURI.empty())
+        {
+            LOG_INF("Remote config url is not specified in coolwsd.xml");
+            return; // no remote config server setup.
+        }
+        else if (!Util::iequal(remoteServerURI.getScheme(), "https"))
+        {
+            LOG_ERR("Remote config url should only use HTTPS protocol");
+            return;
+        }
+
+        //checks certificates
+        try
+        {
+            Poco::Net::Context::Params sslRemoteParams;
+
+            Poco::SharedPtr<Poco::Net::PrivateKeyPassphraseHandler> consoleClientHandler =
+                new Poco::Net::KeyConsoleHandler(false);
+            Poco::SharedPtr<Poco::Net::InvalidCertificateHandler> invalidClientCertHandler =
+                new Poco::Net::AcceptCertificateHandler(false);
+
+            sslRemoteParams.certificateFile = COOLWSD::getPathFromConfigWithFallback(
+                "storage.ssl.cert_file_path", "ssl.cert_file_path");
+            sslRemoteParams.privateKeyFile = COOLWSD::getPathFromConfigWithFallback(
+                "storage.ssl.key_file_path", "ssl.key_file_path");
+            sslRemoteParams.caLocation = COOLWSD::getPathFromConfigWithFallback(
+                "storage.ssl.ca_file_path", "ssl.ca_file_path");
+            sslRemoteParams.cipherList = COOLWSD::getPathFromConfigWithFallback(
+                "storage.ssl.cipher_list", "ssl.cipher_list");
+
+            sslRemoteParams.verificationMode =
+                (sslRemoteParams.caLocation.empty() ? Poco::Net::Context::VERIFY_NONE
+                                                    : Poco::Net::Context::VERIFY_STRICT);
+            sslRemoteParams.loadDefaultCAs = true;
+
+            Poco::Net::Context::Ptr sslClientContext =
+                new Poco::Net::Context(Poco::Net::Context::CLIENT_USE, sslRemoteParams);
+
+            Poco::Net::SSLManager::instance().initializeClient(
```

Comment:
We are initializing Poco SSL, but we are not using Poco sockets? 

`httpSession->syncRequest(request);` doesn't use Poco and so the `Poco::Net::SSLManager` is unnecessary here. Instead, we would want `ssl::Manager::initializeClientContext()`.

See `StorageBase::initialize()` for how both `Poco::Net::SSLManager` and our `ssl::Manager` are initialized. Of course we want to do the initialization once, and we want to have only one function that does that. So perhaps we can move `StorageBase::initialize()` to `COOLWSD::initializeSSL()` and remove the `// checks certificates` try/catch above?

Notice that we want to initialize the SSL layer when either Storage *or* Remote config need SSL.

Right now SSL in Remote config will work because we are reusing the Storage settings. But when/if they are different, it will stop working. And of course the above initialization code is completely unnecessary (I think the call that I highlighted here will check that it is already initialized and return immediately.)